### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by wojtekmach**

### CODE REVIEW

File: lib/db_connection.ex [hunk 1]
```diff
@@ -804,19 +806,20 @@ defmodule DBConnection do
   ### Options
 
   See module documentation. The pool and connection module may support other
-  options. All options are passed to `handle_status/2`.
+  options. All options are passed to `c:handle_status/2`.
 
   ### Example
 
       {:ok, conn, _result} = DBConnection.begin(pool)
```
Comment (wojtekmach) [inline lib/db_connection.ex]: there's neither `DBConnection.begin` nor `DBConnection.commit` below in the example, need to update this.

File: lib/db_connection.ex [hunk 2]
```diff
@@ -134,7 +134,7 @@ defmodule DBConnection do
 
   This callback is called when the control of the state is passed back
   to the connection process. It should reverse any changes made in
-  `checkout/2`.
+  `c:checkout/1`.
```
[the remark this answers is not in the document]
Comment (josevalim) [reply lib/db_connection.ex]: Yes.

File: lib/db_connection.ex (hunk 1 above)
Comment (josevalim) [reply lib/db_connection.ex]: It should be `DBConnection.transaction`.

**Comment by josevalim** [comment]
:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

File: lib/db_connection.ex (hunk 1 above)
Comment (florinpatrascu) [reply lib/db_connection.ex]: Hi there - without `begin`/`commit`, how would one use non-closure based transactions, with the new [DBConnection.html.transaction/3](
https://hexdocs.pm/db_connection/2.0.0-rc.0/DBConnection.html#transaction/3)? This is a follow up to #87, which is closed now. Thank you!

Comment (josevalim) [reply lib/db_connection.ex]: It is not possible to have non-closure based transactions. It adds to many complications and possible failure scenarios, so we decided to not support it.

**Pull request merged**